chameleon/src/test/java/org/hibernate/omm/crud/SimpleCRUDTests.java

```diff
@@ -30,21 +30,23 @@ void tearDown() {
 	}
 
 	Book insertBook() {
-		return getSessionFactory().fromSession( session -> {
-			var book = new Book();
-			book.id = id;
-			book.title = "War and Peace";
-			book.author = "Leo Tolstoy";
-			book.publishYear = 1869;
-			session.persist( book );
-			return book;
-		} );
+		var book = new Book();
+		book.id = id;
+		book.title = "War and Peace";
+		book.author = "Leo Tolstoy";
+		book.publishYear = 1869;
+
+		getSessionFactory().inTransaction( session -> session.persist( book ));
+		return book;
 	}
 	void deleteBook() {
-		getSessionFactory().inTransaction( session -> {
-			var book = session.getReference( Book.class,  id );
-			session.remove( book );
-		} );
+		try {
+			getSessionFactory().inTransaction( session -> {
+				var book = session.getReference( Book.class,  id );
+				session.remove( book );
+			} );
+		} catch (Exception ignored) {
```

Comment:
This was throwing when the book wasn't in the db.  There's probably a better way than catching all exceptions though...

Reply:
This is simple bug. I would hesitate to swallow such defect for exposing issue and fixing it might be better. The issue has been solved by using HQL instead.